zebrad/tests/common/lightwalletd/wallet_grpc_test.rs

```diff
@@ -344,5 +343,46 @@ pub async fn run() -> Result<()> {
     // Make sure the subversion field is zebra the user agent
     assert_eq!(lightd_info.zcashd_subversion, USER_AGENT);
 
+    // Transactions in mempool
+
+    let transaction = Transaction::zcash_deserialize(&zebra_test::vectors::DUMMY_TX3[..]).unwrap();
+    let transaction_bytes = transaction.zcash_serialize_to_vec().unwrap();
+
+    // Send transaction by calling `SendTransaction`
+    let request1 = RawTransaction {
+        data: transaction_bytes,
+        height: -1,
+    };
+    let _ = rpc_client.send_transaction(request1).await?.into_inner();
+
+    // Wait a bit to query the mempool
+    tokio::time::sleep(std::time::Duration::from_secs(10)).await;
+
+    // Call `GetMempoolTx` and get a stream of transactions
+    let mut transactions_stream = rpc_client
+        .get_mempool_tx(Exclude { txid: vec![] })
+        .await?
+        .into_inner();
+
+    // Make sure our transaction, and only our transaction was inserted to the mempool.
+    let mut counter = 0;
+    while let Some(tx) = transactions_stream.message().await? {
+        assert_eq!(tx.hash, transaction.hash().bytes_in_display_order());
+        counter += 1;
+    }
+
+    assert_eq!(counter, 1);
+
+    // Get the mempool by calling `GetMempoolStream`.
+    let mut transaction_stream = rpc_client.get_mempool_stream(Empty {}).await?.into_inner();
+
+    let mut counter = 0;
+    while let Some(_tx) = transaction_stream.message().await? {
+        counter += 1;
+    }
+
+    // This one is not working.
+    assert_eq!(counter, 0);
```

Comment:
I am not sure yet if this is a problem in our end or in lightwalletd. I will check after the other call gets sorted but it could be because of https://github.com/adityapk00/lightwalletd/blob/master/frontend/service.go#L515-L517

I tried this call instead of `GetMempoolTx` just to be sure that the stream was not consumed by the first call, however this didn't worked.

Reply:
Good find!

I think it's ok for the test to expect no transactions for now. We can link to the `lightwalletd` code that disables the mempool stream, and fix the test when `lightwalletd` starts streaming transactions again.